wurst/lib/Lodash.wurst

```diff
@@ -305,6 +316,341 @@ public function each<T>(VoidFunction<T> func, LinkedList<T> list)
 public function LinkedList<T>.each<T>(VoidFunction<T> func)
     each(func, this)
 
+public function zipObject<A, B>(LinkedList<A> a, LinkedList<B> b) returns IterableMap<A, B>
+    let map = new IterableMap<A, B>()
+    let iterA = a.iterator()
+    let iterB = b.iterator()
+    while iterA.hasNext()
+        let val =  iterB.hasNext()
+            ? iterB.next()
+            : null
+        map.put(iterA.next(), val)
+    iterA.close()
+    iterB.close()
+    a.maybeFree()
+    b.maybeFree()
+    return map
+
+public function LinkedList<A>.zipObject<A, B>(LinkedList<B> lst) returns IterableMap<A, B>
+    return zipObject(this, lst)
+
+public class Pair<A, B>
+    A a
+    B b
+
+    construct(A a, B b)
+        this.a = a
+        this.b = b
+
+public function pair<A, B>(A a, B b) returns Pair<A, B>
+    return new Pair(a, b)
+
+public function zip<A, B>(LinkedList<A> a, LinkedList<B> b) returns LinkedList<Pair<A, B>>
+    let lst = new LinkedList<Pair<A, B>>()
+    let iterA = a.iterator()
+    let iterB = b.iterator()
+    while iterA.hasNext() and iterB.hasNext()
+        lst.add(pair(iterA.next(), iterB.next()))
+    iterA.close()
+    iterB.close()
+    a.maybeFree()
+    b.maybeFree()
+    return lst
+
+public function LinkedList<A>.zip<A, B>(LinkedList<B> lst) returns LinkedList<Pair<A, B>>
+    return zip(this, lst)
+
+public class Range
+    let incr = 1
+    let start = 0
+    let finish = INT_MAX
+    var curr = start
+
+    construct()
+
+    construct(int start, int finish, int incr)
+        this.start = start
+        this.finish = finish
+        this.incr = incr
+        curr = start
+
+    function hasNext() returns bool
+        return incr > 0 ? curr < finish : curr > finish
+
+    function next() returns int
+        let ret = curr
+        curr += incr
+        return ret
+
```

Comment:
is hasNext() supposed to be called manually before using next()? Looks like this can go over the upper bound if being called without hasNext()

Reply:
@Marsunpaisti `for` loops always call `hasNext()` before `next()`